src/core/cgroup.c

```diff
@@ -1361,6 +1361,22 @@ int unit_attach_pids_to_cgroup(Unit *u) {
         return 0;
 }
 
+static void cgroup_xattr_apply(Unit *u) {
+        int r;
+
+        assert(u);
+
+        if (!MANAGER_IS_SYSTEM(u->manager))
+                return;
+
+        if (sd_id128_is_null(u->invocation_id))
+                return;
+
+        r = cg_set_xattr(SYSTEMD_CGROUP_CONTROLLER, u->cgroup_path, "trusted.invocation_id", &u->invocation_id, sizeof(u->invocation_id), 0);
```

Comment:
I wonder if it wouldn't be nicer to store the xattr in text form? It would make introspecting by users easier.